### PR TITLE
[W06H01] Use RegExp instead of simple keyword search for `testUsageOfLoops`

### PR DESCRIPTION
**Pull request by nilsreichardt**
Using regular expression instead of a simple keyword search (suggested by @julian-hartl in https://github.com/MaximilianAnzinger/pgdp2223-tests/pull/114/files#r1035113758).

### CODE REVIEW

File: w06h01/test/pgdp/LoopTest.java [hunk 1]
```diff
@@ -22,7 +24,12 @@ public void testUsageOfLoops() {
                 "src/pgdp/datastructures/lists/RecIntListElement.java"
         };
 
-        String[] notAllowedKeywords = new String[] { "for", "while", "Stream" };
+        String[] notAllowedRegExp = new String[] {
+                "for\s*[(]{1}",
+                "while\s*[(]{1}",
+                "Stream.of",
+                "Stream<",
```
Comment (nilsreichardt) [inline w06h01/test/pgdp/LoopTest.java:31]: There are still thousands ways to use Streams. However, I think for the beginning is this enough

Comment (TheAleWu) [reply w06h01/test/pgdp/LoopTest.java:31]: I think "Arrays.stream(" would be a good candidate too as we already got to know the Arrays class in the past

File: w06h01/test/pgdp/LoopTest.java [hunk 2]
```diff
@@ -36,9 +43,11 @@ public void testUsageOfLoops() {
             String filteredFile = removeMultipleMethods(file, allowedMethods);
 
             // Check if there are any loops
-            for (String keyword : notAllowedKeywords) {
-                assertTrue(!filteredFile.contains(keyword),
-                        "You are not allowed to use loops in your solution! Found a loop : " + keyword);
+            for (String regExp : notAllowedRegExp) {
+                Pattern pattern = Pattern.compile(regExp);
+                Matcher matcher = pattern.matcher(filteredFile);
+                assertTrue(!matcher.find(),
```
Comment (TheAleWu) [inline w06h01/test/pgdp/LoopTest.java:49]: assertFalse(.., ..) would be a bit more readable and understandable than asserting for a negated false value.

**Review by TheAleWu — CHANGES_REQUESTED** [review]

File: w06h01/test/pgdp/LoopTest.java (hunk 2 above)
Comment (nilsreichardt) [reply w06h01/test/pgdp/LoopTest.java]: oh yes 👍

File: w06h01/test/pgdp/LoopTest.java (hunk 1 above)
Comment (nilsreichardt) [reply w06h01/test/pgdp/LoopTest.java]: Added ✅

**Review by julian-hartl — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**